### PR TITLE
Drift in config.hcl

### PR DESCRIPTION
**Pull request by disq**
- `drift init` is gone, config is now in `config.hcl`
- `--account-ids` param is now in `config.hcl`, `--module-config` is gone. Resource configs can now be overridden in `config.hcl` (just add a `modules.drift.provider` block with the correct label)
- Multiple module config profiles are supported, if single one then it's auto-selected, or specify one using `--profile <name>`
- Embedded drift config lives under `/pkg/module/drift/`, `/modules` is gone
- Type based, id based and tag based resource ignoring and whitelisting is supported. `skip_resources` is renamed to `ignore_resources` to comply better with the change.
- Tag filtering is done only on cloud resources, IaC resources are not filtered. All specified tags should exist on the resource, but extra tags on the cloud side are allowed. To do a boolean OR, multiple items describing same resource type can be added to the check/ignore lists.
- S3 or local file TFstates with multiple files/objects are supported. Can still use one or more args to `drift scan` to point it to local file(s) without editing config.hcl
- `cloudquery init <providers>` now generates a module block in the config as well, iterating through all modules and asking for their example configs.

Example module config:
```
// Module Configurations
modules {
  // drift configuration block
  drift "drift-example" {
    // state block defines from where to access the state
    terraform {
      backend  = "local"
#      backend  = "s3"
      files = [ "/path/to/terraform.json" ]
#      bucket   = "<terraform state bucket>"
#      keys     = ["<terraform state key>"]
#      region   = "us-east-1"
#      role_arn = ""
    }

    provider "aws" {
#      account_ids      = ["123456789"]
      check_resources = [
#        "ec2.instances:*",
#        "ec2.instances:[Type=integration_test,TestId=integration]",
        "*:[Type=integration_test,TestId=integration]",
        "*:[TestId=windowsfifl5fe]",
#        "cloudwatchlogs.filters:*"
      ]
      ignore_resources   = [
#        "ec2.instances:*",
#        "ec2.instances:i-0e7b9c38956f77f0e",
#        "ec2.instances:[Type=integration_test]",
      ]
  }
}
```

### CODE REVIEW

**Review by roneli — APPROVED** [review]
Some changes required, but looks good all in all. I would add that the config.go looks a little complex and scattered, but again i'd postpone it for a later refactor.

File: pkg/config/config_parser.go [hunk 1]
```diff
@@ -81,6 +81,9 @@ func (p *Parser) decodeConfig(body hcl.Body, diags hcl.Diagnostics) (*Config, hc
 			if cfg != nil {
 				config.Providers = append(config.Providers, cfg)
 			}
+		case "modules":
+			// Module manager will process this for us
+			config.Modules = block.Body
```
[the remark this answers is not in the document]
Comment (disq) [reply pkg/config/config_parser.go:86]: the problem is config is parsed at one point and there's also the issue of method ownership (which method is where, console client vs cq client and so on) so to parse it before module manager is initialized (not sure about this) might be problematic.

File: pkg/module/types.go [hunk 2]
```diff
@@ -13,9 +13,11 @@ type Module interface {
 	// ID returns the name of the module
 	ID() string
 	// Configure configures the module to run
-	Configure(context.Context, hcl.Body) error
+	Configure(context.Context, map[string]hcl.Body, interface{}) error
```
[the remark this answers is not in the document]
Comment (disq) [reply pkg/module/types.go]: Fixed

File: pkg/module/drift/drift.go [hunk 3]
```diff
@@ -48,21 +55,42 @@ func (d *Drift) ID() string {
...
-	p := NewParser("")
+func (d *Drift) Configure(ctx context.Context, profileConfig map[string]hcl.Body, runParams interface{}) error {
+	d.params = runParams.(RunParams)
 
-	cfg, diags := p.Decode(config, nil)
-	if diags.HasErrors() {
-		return diags
+	if d.params.Profile == "" && len(profileConfig) > 1 {
+		return fmt.Errorf("multiple drift profiles detected, choose one with --profile")
+	}
```
[the remark this answers is not in the document]
Comment (disq) [reply pkg/module/drift/drift.go:74]: Maybe after we have more than a single module we can take a look at where this is handled.

File: pkg/module/drift/config.go [hunk 4]
```diff
@@ -38,6 +46,43 @@ type ResourceConfig struct {
...
+
+	// check/ignore selectors
+	acl ResourceACL
+}
+
+type ResourceACL struct {
+	CheckEnabled bool
+	Check        ResourceSelectors
+	Ignore       ResourceSelectors
+}
```
[the remark this answers is not in the document]
Comment (disq) [reply pkg/module/drift/config.go]: Done

File: pkg/module/drift/terraform.go [hunk 5]
```diff
@@ -161,7 +161,7 @@ func parseTerraformAttribute(val interface{}, t schema.ValueType) interface{} {
 	}
 }
 
-func driftTerraform(ctx context.Context, logger hclog.Logger, conn *pgxpool.Conn, cloudName string, cloudTable *traversedTable, resName string, resources map[string]*ResourceConfig, iacData *IACConfig, states TFStates, runParams RunParams) (*Result, error) {
+func driftTerraform(ctx context.Context, logger hclog.Logger, conn *pgxpool.Conn, cloudName string, cloudTable *traversedTable, resName string, resources map[string]*ResourceConfig, iacData *IACConfig, states TFStates, runParams RunParams, accountIDs []string) (*Result, error) {
```
[the remark this answers is not in the document]
Comment (disq) [reply pkg/module/drift/terraform.go:164]: shouldn't get any longer than this... maybe could skip ctx/logger or at least combine them in the future.

**Pull request merged**